### ISSUE TITLE
fix: Update to `class-validator@0.13.2` generates `error TS7016: Could not find a declaration file for module 'validator'.`

### ISSUE THREAD

**Issue by KeithGillette** [typestack/class-validator#1412]
### Description
Updating a project from `class-validator@0.13.1` to `class-validator@0.13.2` generates 16 `error TS7016: Could not find a declaration file for module 'validator'.` messages on `tsc` transpilation when `tsconfig.json` is configured with the default `skipLibCheck: false`.

**Minimal code-snippet showcasing the problem**
[`class-validator@0.13.2` removes `@types/validator` from the `package.json` `dependencies`](https://github.com/typestack/class-validator/commit/5f91937ddbf9d4247477d8c5993834d52d4c6294#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L39), which now requires consuming applications to manually add `@types/validator` to avoid `error TS7016` on transpilation when the default `skipLibCheck: false`.

### Expected behavior
Update from `0.13.1` to `0.13.2` should allow project to build without error.

### Actual behavior

```
node_modules/class-validator/types/decorator/string/IsAlpha.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsAlphanumeric.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsCurrency.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '/Users/keith.gillette/Code/TaskTrain/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsDateString.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsDecimal.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsEmail.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsFQDN.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsHash.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsIdentityCard.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsISO8601.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsISSN.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsMacAddress.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsMobilePhone.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsNumberString.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsPostalCode.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~

node_modules/class-validator/types/decorator/string/IsUrl.d.ts:2:25 - error TS7016: Could not find a declaration file for module 'validator'. '~/node_modules/validator/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/validator` if it exists or add a new declaration (.d.ts) file containing `declare module 'validator';`

2 import ValidatorJS from 'validator';
                          ~~~~~~~~~~~


Found 16 errors.
```

**Comment by Sonahit** [comment]
Same error after updating

**Comment by kliuiev-io** [comment]
Fixes locally by npm i --save-dev @types/validator but it's still a bug

**Comment by angelxmoreno** [comment]
@kliuiev-io @KeithGillette which do you advise

1. sticking with 0.13.1
2. installing @types/validator

**Comment by ianhowe76** [comment]
I am guessing this is related to `@types/validator` moving to `devDependancies` in v1.13.2

https://github.com/typestack/class-validator/compare/v0.13.1...v0.13.2

**Comment by DnOberon** [comment]
Really don't want to sticky my project to the previous version, any movement on this?

**Comment by SchroederSteffen** [comment]
Considering that `@types` packages are only required for the build, do we agree that moving it to the dev dependencies is actually the right call? 🤔 
It would come down to a documentation issue like: "You need to `npm i -D @types/validator` yourself."

**Comment by KeithGillette** [comment]
While installing `@types/validator` as a project development dependency will fix the transpilation error messages, since this is a published package that is consumed by other applications, I'm more in agreement with [this StackOverflow answer](https://stackoverflow.com/a/46011417/7443584) supporting the previous practice of keeping `@types/validator` in dependencies so that it's available during development without that extra step.

**Comment by mohd-akram** [comment]
Any update on this? This was already fixed once [before](https://github.com/typestack/class-validator/commit/a39c7ac452c412b6a728193968eb410fc40766f7) in the project's history.

**Comment by flmuel** [comment]
please fix and merge :)

**Comment by NoNameProvided** [comment]
Fixed in https://github.com/typestack/class-validator/commit/12f8d86d8507563dfb8e53b6c2e7c490c08e121e

**Comment by nVitius** [comment]
@NoNameProvided Can we expect a new release that includes that fix?

**Comment by NoNameProvided** [comment]
Yes

**Comment by NoNameProvided** [comment]
This feature/fix has been released in `0.14.0`.
